tenable/io/v3/assets/api.py

```diff
@@ -0,0 +1,403 @@
+                                   ('test', 'oper', '2')
+                            ),
+                    'and', ('test', 'oper', 3)
+                   )
+                    >>> {'or': [
+                    {'and': [
+                        {'value': '1', 'operator': 'oper', 'property': '1'},
+                        {'value': '2', 'operator': 'oper', 'property': '2'}
+                        ]
+                    }],
+                    'and': [
+                        {'value': '3', 'operator': 'oper', 'property': 3}
+                        ]
+                    }
+
+                As the filters may change and sortable fields may change over
+                time, it's highly recommended that you look at the output of
+                the :py:meth:`tio.v3.vm.filters.asset_filters()`
+                endpoint to get more details.
+            sort list(tuple, Dict):
+                A list of dictionaries describing how to sort the data
+                that is to be returned.
+
+                Examples:
+                    - ``[("field_name_1", "asc"),
+                             ("field_name_2", "desc")]``
+                    - ``[{'property': 'last_observed', 'order': 'desc'}]``
+            limit (int):
+                Number of objects to be returned in each request.
+                Default is 1000.
+            next (str):
+                The pagination token to use when requesting the next page of
+                results.  This token is presented in the previous response.
+            return_resp (bool):
+                If set to true, will override the default behavior to return
+                an iterable and will instead return the results for the
+                specific page of data.
+            return_csv (bool):
+                If set to true, It wil return the CSV Iterable. Returns all
+                data in text/csv format on each next call with row headers
+                on each page.
+
+        Returns:
+            Returns:
+                Iterable:
+                    The iterable that handles the pagination and potentially
+                    async requests for the job.
+                requests.Response:
+                    If ``return_json`` was set to ``True``, then a response
+                    object is instead returned instead of an iterable.
+
+        Examples:
+            >>> tio.v3.assets.search_assets(filter=('netbios_name', 'eq',
+            >>>  'SCCM'), fields=["name", "netbios_name", "last_login"],
+            >>>    limit=2, sort=[('last_observed', 'asc')])
+        '''
+        search_api_path = 'api/v3/assets/search'
+        return_csv = kw.pop('return_csv')
+        return super().search(
+            resource='assets',
+            iterator_cls=AssetCSVIterator if return_csv else
+            AssetSearchIterator,
+            is_sort_with_prop=False,
+            api_path=search_api_path,
+            **kw
+        )
```

Comment:
This could be re-written to be be more clear.

```suggestion
        iclass = AssetSearchIterator
        if kw.pop('return_csv', False):
            iclass = AssetCSVIterator
        return super().search(resource='assets',
                              iterator_cls=iclass,
                              is_sort_with_prop=False,
                              api_path=f'{self._path}/search',
                              **kw
                              )
```